python/opae.admin/opae/admin/tools/fpgasupdate.py

```diff
@@ -351,27 +381,183 @@ def canonicalize_bdf(bdf):
     return None
 
 
+def ioctl_status(sec_dev):
+    # struct fpga_image_status {
+	# /* Output */
+	# __u32 remaining_size;			/* size remaining to transfer */
+	# enum fpga_image_prog progress;		/* current phase of image load */
+	# enum fpga_image_prog err_progress;	/* progress at time of error */
+	# enum fpga_image_err err_code;		/* error code */
+    # };
+    status_header = array.array('B', [0] * 32)
+    fcntl.ioctl(sec_dev, IOCTL_FPGA_IMAGE_LOAD_STATUS, status_header, True)
```

Comment:
Need a try: here to guard against ioctl exceptions.  If the ioctl() fails, an OSError exception is raised.